EpiAware/src/EpiAwareUtils/prefix_submodel.jl

```diff
@@ -0,0 +1,30 @@
+@doc raw"
+Generate a submodel with an optional prefix. A lightweight wrapper around the `@submodel` macro from DynamicPPL.jl.
+
+# Arguments
+
+- `model::AbstractModel`: The model to be used.
+- `fn::Function`: The Turing @model function to be applied to the model.
+- `prefix::String`: The prefix to be used. If the prefix is an empty string, the submodel is created without a prefix.
+
+# Returns
+
+- `submodel`: The returns from the submodel are passed through.
+
+# Examples
+
+```julia
+using EpiAware
+submodel = prefix_submodel(CombineLatentModels([FixedIntercept(0.1), AR()]), generate_latent, \"Test\", 10)
+rand(submodel)
+```
+"
+@model function prefix_submodel(
```

Comment:
```suggestion
@model function prefix_submodel(
        model::AbstractModel, fn::Function, prefix::String = ""; kwargs...)

        @submodel prefix=eval(prefix) submodel=fn(model, kwargs...)
        
    return submodel
end
```

Would seem to be more idomatic, does this cause problems with `eval`?

Reply:
This is because if you supply anything at all to `prefix` as part of submodule it changes in name. So `""` produces `.original-var` for example.

I think it is debatable if we want to support empty prefixing here as you should never not have a prefix (and I assume this takes some compute. Going to move to an issue.

Reply:
See #309